components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -619,9 +637,35 @@ static int esp_websocket_client_send_with_exact_opcode(esp_websocket_client_hand
 
 esp_websocket_client_handle_t esp_websocket_client_init(const esp_websocket_client_config_t *config)
 {
-    esp_websocket_client_handle_t client = calloc(1, sizeof(struct esp_websocket_client));
+    int memory_type = 0;
+    if ((config->memory_type & ~VALID_MEMORY_FLAGS) != 0) {
+        ESP_LOGE("TAG", "Invalid memory_type flags set.\n");
+        return NULL;
+    } else if (config->memory_type == 0) {
+        memory_type = MALLOC_CAP_DEFAULT; //MALLOC_CAP_DEFAULT
+        ESP_LOGD("TAG","memory_type is not set, using MALLOC_CAP_DEFAULT memory [%d] versus external [%d].\n",memory_type,MALLOC_CAP_SPIRAM);
+    } else {
+        memory_type = config->memory_type;
+        ESP_LOGD("TAG","memory_type is set to %d \n", config->memory_type);
+    }
+
+    esp_websocket_client_handle_t client = heap_caps_calloc(1, sizeof(struct esp_websocket_client),memory_type);
     ESP_WS_CLIENT_MEM_CHECK(TAG, client, return NULL);
 
+    if (config->core){
```

Comment:
I would prefer it to be introduced in a separated PR. Since we'll have to take into consideration portability among different IDF versions given the proposal of usage `xTaskCreatePinnedToCoreWithCaps`. Could you please move this to a second PR?